osu.Game.Tests/Visual/Gameplay/TestSceneScoring.cs

```diff
@@ -245,41 +276,42 @@ private void runForAlgorithm(string name, Color4 colour, Action applyHit, Action
             for (int i = 0; i < maxCombo; i++)
             {
                 if (graphs.MissLocations.Contains(i))
-                    applyMiss();
+                    scoringAlgorithm.ApplyMiss();
                 else if (graphs.NonPerfectLocations.Contains(i))
-                    applyNonPerfect();
+                    scoringAlgorithm.ApplyNonPerfect();
                 else
-                    applyHit();
+                    scoringAlgorithm.ApplyHit();
 
-                results.Add(getTotalScore());
+                results.Add(scoringAlgorithm.GetTotalScore());
             }
 
-            graphs.Add(new LineGraph
+            LineGraph graph;
+            graphs.Add(graph = new LineGraph
             {
-                Name = name,
+                Name = scoringAlgorithm.Name,
                 RelativeSizeAxes = Axes.Both,
-                LineColour = colour,
+                LineColour = scoringAlgorithm.Colour,
                 Values = results
             });
 
-            legend.Add(new OsuSpriteText
-            {
-                Colour = colour,
-                RelativeSizeAxes = Axes.X,
-                Width = 0.5f,
-                Text = $"{FontAwesome.Solid.Circle.Icon} {name}"
-            });
-
-            legend.Add(new OsuSpriteText
+            legend.Add(new LegendEntry(scoringAlgorithm, graph)
             {
-                Colour = colour,
-                RelativeSizeAxes = Axes.X,
-                Width = 0.5f,
-                Text = $"final score {getTotalScore():#,0}"
+                AccentColour = scoringAlgorithm.Colour,
             });
         }
     }
 
+    public class ScoringAlgorithm
```

Comment:
This is kinda fast-and-loose, especially with the bindable, but I figure it's a test and I didn't want to hem and haw over this _too_ much. Will change on request though.